gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/architecture/LegacyUsageTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package architecture;
+
+import static com.tngtech.archunit.base.DescribedPredicate.not;
+import static com.tngtech.archunit.core.domain.JavaClass.Predicates.resideInAnyPackage;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class LegacyUsageTest extends AbstractApimArchitectureTest {
+
+    /**
+     * Code from {@link AbstractApimArchitectureTest#GRAVITEE_APIM_PACKAGE} should not depend on class from legacy code in {@link AbstractApimArchitectureTest#GRAVITEE_REST_API_SERVICE_PACKAGE}
+     */
+    @Test
+    public void should_not_use_legacy_code() {
```

Comment:
Classes in "core" package should not use legacy code. However, we have to authorize that "infra" classes can use legacy code; otherwise, we would have too much to rewrite 🙈

Reply:
I agree. removed this class as it is now addressed by `CoreRulesTest`